fdbserver/DataDistribution.actor.cpp

```diff
@@ -2569,6 +2578,186 @@ ACTOR Future<Void> waitUntilHealthy(DDTeamCollection* self, double extraDelay =
 	}
 }
 
+// This function is very expensive to call. Use with caution.
+ACTOR Future<Void> printSnapshotTeamsInfo(Reference<DDTeamCollection> self, std::string tcIdString) {
+	state DatabaseConfiguration configuration;
+	state std::map<UID, Reference<TCServerInfo>> server_info;
+	state ServerStatusMap server_status;
+	state vector<Reference<TCTeamInfo>> teams;
+	state std::map<Standalone<StringRef>, Reference<TCMachineInfo>> machine_info;
+	state std::vector<Reference<TCMachineTeamInfo>> machineTeams;
+	state std::vector<std::string> internedLocalityRecordKeyNameStrings;
+	state int machineLocalityMapEntryArraySize;
+	state std::vector<Reference<LocalityRecord>> machineLocalityMapRecordArray;
+	state int traceEventsPrinted = 0;
+	state std::vector<const UID*> uids;
+	loop {
+		wait(self->printDetailedTeamsInfo.onTrigger() || delay(SERVER_KNOBS->DD_TEAMS_INFO_PRINT_INTERVAL));
+
+		traceEventsPrinted = 0;
+		if (now() - self->lastTeamsInfoSnapshotTime >= SERVER_KNOBS->DD_TEAMS_INFO_SNAPSHOT_REFRESH_INTERVAL) {
+			// Refresh state snapshot
+			self->lastTeamsInfoSnapshotTime = now();
+
+			configuration = self->configuration;
+			server_info = self->server_info;
+			teams = self->teams;
+			machine_info = self->machine_info;
+			machineTeams = self->machineTeams;
+			internedLocalityRecordKeyNameStrings = self->machineLocalityMap._keymap->_lookuparray;
+			machineLocalityMapEntryArraySize = self->machineLocalityMap.size();
+			machineLocalityMapRecordArray = self->machineLocalityMap.getRecordArray();
+			std::vector<const UID*> _uids = self->machineLocalityMap.getObjects();
+			uids = _uids;
+
+			auto const& keys = self->server_status.getKeys();
+			for (auto const& key : keys) {
+				server_status.set(key, self->server_status.get(key));
+			}
+		}
+
+		// Print to TraceEvents
+		TraceEvent("DDConfig")
+		    .detail("StorageTeamSize", configuration.storageTeamSize)
+		    .detail("DesiredTeamsPerServer", SERVER_KNOBS->DESIRED_TEAMS_PER_SERVER)
+		    .detail("MaxTeamsPerServer", SERVER_KNOBS->MAX_TEAMS_PER_SERVER)
+		    .detail("TeamCollectionID", tcIdString);
+
+		TraceEvent("ServerInfo").detail("Size", server_info.size()).detail("TeamCollectionID", tcIdString);
+		state int i = 0;
+		state std::map<UID, Reference<TCServerInfo>>::iterator server = server_info.begin();
```

Comment:
Can you refactor the `traceServerInfo()`  so that it can be used here as well?

The same goes for the other existing functions, such as `traceServerTeamInfo()`, used in `traceAllInfo()`